include/swift/AST/Evaluator.h

```diff
@@ -224,6 +231,38 @@ class Evaluator {
   /// so all clients must cope with cycles.
   llvm::DenseMap<AnyRequest, std::vector<AnyRequest>> dependencies;
 
+  /// A stack of dependency sources in the order they were evaluated.
```

Comment:
Presumably LIFO? Why is the LIFOness important? Or is it?

Reply:
It's not important. It's just the most efficient representation of the problem domain.